homeassistant/helpers/event.py

```diff
@@ -388,24 +390,22 @@ def _async_track_event(
 
     hass_data = hass.data
 
-    callbacks: dict[
-        str, list[HassJob[[EventType[_TypedDictT]], Any]]
-    ] | None = hass_data.get(callbacks_key)
-    if not callbacks:
+    callbacks: dict[str, list[HassJob[[EventType[_TypedDictT]], Any]]] | None
+    if not (callbacks := hass_data.get(callbacks_key)):
         callbacks = hass_data[callbacks_key] = {}
 
     if listeners_key not in hass_data:
         hass_data[listeners_key] = hass.bus.async_listen(
             event_type,
             ft.partial(dispatcher_callable, hass, callbacks),
             event_filter=ft.partial(filter_callable, hass, callbacks),
+            run_immediately=run_immediately,
```

Comment:
This feels weird, as now it is possible to pass `listeners_key` twice to `_async_track_event` and only the first invocation will define the `run_immediately` value.

Reply:
I think we have the same issue with `dispatcher_callable` and `filter_callable` and `event_type`

I'll add a comment to the doc string

Reply:
```
    """Track an event by a specific key.
    
    This function is intended for internal use only.

    The `dispatcher_callable`, `filter_callable`, `event_type`, and `run_immediately`
    must always be the same for the `listener_key` as the first call to this
    function will set the `listener_key` in hass.data.
    """
```

We should probably redesign it in the future to be harder to get wrong but since its currently only used in a very specific way its not a problem right now

Reply:
aea1648db52c1206f32111536755fac3a81a11e8

Reply:
Refactored in https://github.com/home-assistant/core/pull/111150 as a follow up